### PR TITLE
[Server] MeshSync fingerprinting support.

### PR DESCRIPTION
**Pull request by MUzairS15**
1. The PR updates the Dashboard K8s resources table to use fingerprint data (icons and colour).
2. Adds the ability to upgrade discovered resources by MeshSync to a Meshery Connection by completing the registration flow.
3. Adds Connection and Credential definition for Prometheus and Grafana.
4. Updates existing APIs Prometheus and Grafana to use Connection constructs.

To add support for the registration of any other resources as  Meshery Connection, the following process needs to be followed.
1. Add Connection and credential definition at the appropriate location (in the respective models directory).
2. Implement the `Action` interface. The implementation is required for the `verify` step during the registration process. (Prerequisites and reachability check)
A default implementation of the interface is also provided for particular states. The connections registered as part of this process are not managed in real-time and are more of an administrative setup.

When the registration process starts, the server responds with the schemas and an identifier, since the registration process is not time-bound and there could be multiple registrations at the same time, the identifier is used to track the registration process.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
@MUzairS15 are these failures expected?

**Comment by MUzairS15** [comment]
Meshsync package needs upgraded. Since a new meshsync is released I will upgrade.

File: ui/components/configuratorComponents/MeshModel/NodeIcon.js
Comment (MUzairS15) [inline ui/components/configuratorComponents/MeshModel/NodeIcon.js:1]: Sourced from extensions.
Needs to be deleted from extensions and injected as a remote component.

File: ui/components/configuratorComponents/MeshModel/DesignerDrawerNodeShapes.js
Comment (MUzairS15) [inline ui/components/configuratorComponents/MeshModel/DesignerDrawerNodeShapes.js:1]: Sourced from extensions.
Needs to be deleted from extensions and injected as a remote component.

**Review by leecalcote — CHANGES_REQUESTED** [review]
@MUzairS15 build and lint checks are still failing. Let me know when the is ready for review.

**Comment by MUzairS15** [comment]
Screenshot 2023-11-27 at 4 12 24 PM
On local UI lint and Meshkit error utility passes.

**Pull request merged**